app/assets/javascripts/settings.coffee

```diff
@@ -13,7 +15,17 @@ clamp = (min, max, val) ->
 
 class Settings
 
-  locale: "en_us"
+  constructor: ->
+    @locale = navigator.language.replace('-', '_').toLowerCase()
+
+    exactMatch = locales.find((locale) => @locale is locale.code)
+
+    if exactMatch
+      @locale = exactMatch.code
+    else
+      matchingLocale = locales.find((locale) => @locale.startsWith(locale.code.split('_')[0]))
```

Comment:
Because we control the `locales` data, I suggest we pre-split out a new field for `languageCode` in that data instead of doing the split programmatically here.